test/mapping/attractor_mapping_tests.jl

```diff
@@ -74,12 +74,12 @@ function test_basins(ds, u0s, grid, expected_fs_raw, featurizer;
         test_basins_fractions(mapper; known = true, err = aerr)
     end
 
-    @testset "Recurrences" begin
-        mapper = AttractorsViaRecurrences(ds, grid; diffeq, show_progress = false, kwargs...)
+    @testset "Recurrences Non-Sparse" begin
+        mapper = AttractorsViaRecurrences(ds, grid; sparse=false, diffeq, show_progress = false, kwargs...)
```

Comment:
Since the non sparse and sparse version have identical code, and the only thing that changes is an internal array represenrtation, we shouldnt have to test the non sparse version extensively. I propose that you remove the non sparse from these large, detailed tests, and just add a new test elsewere that computes some basins fractions with sparse and non sparse and tests that the result is identical between the two.